src/components/IsUnlocked.js

```diff
@@ -102,23 +109,67 @@ class IsUnlocked extends Component<Props, State> {
       incorrectPassword: null,
     }))
 
-  handleSubmit = async (e: SyntheticEvent<HTMLFormElement>) => {
-    e.preventDefault()
+  unlockDb = async password => {
+    const { fetchAccounts } = this.props
+    const isAccountsDecrypted = await db.hasBeenDecrypted('app', 'accounts')
 
-    const { unlock, fetchAccounts } = this.props
-    const { inputValue } = this.state
+    if (isAccountsDecrypted) {
+      return db.isEncryptionKeyCorrect('app', 'accounts', password)
+    }
 
-    const isAccountsDecrypted = await db.hasBeenDecrypted('app', 'accounts')
     try {
-      if (!isAccountsDecrypted) {
-        await db.setEncryptionKey('app', 'accounts', inputValue.password)
-        await fetchAccounts()
-      } else if (!db.isEncryptionKeyCorrect('app', 'accounts', inputValue.password)) {
-        throw new PasswordIncorrectError()
+      await db.setEncryptionKey('app', 'accounts', password)
+      await fetchAccounts()
+      return true
+    } catch (error) {
+      return false
+    }
+  }
+
+  unlockLibcore = async password => {
+    const { setLibcorePassword } = this.props
+
+    await setLibcorePassword(password)
+    const unlockedWithPW = await isLibcoreUnlocked()
+
+    // Migration time!
+    if (!unlockedWithPW) {
+      // since we know the password is valid for db, if it fails for libcore
+      // it can be because of being from a previous version which didn't implement encryption,
+      // so let's try without a password
+      await setLibcorePassword('')
+      // await reloadLibcore()
+      const unlockedWithoutPW = await isLibcoreUnlocked()
+
+      if (unlockedWithoutPW) {
+        // Libcore is indeed unencrypted, so let's encrypt with db password
+        await changeLibcorePassword('', password)
+        // And reload with this new password
+        await setLibcorePassword(password)
+      } else {
+        // Something is wrong, let's set the password and reset libcore, so that
+        // it creates a brand new db encrypted with the current password
+        await setLibcorePassword(password)
+        resetLibcore()
```

Comment:
this is async, we don't wait it?

Reply:
No logic in this part rely on the completion of the function, so I didn't think it was necessary to await for it, but that doesn't seem to be a good practice